test/valgrind.supp

```diff
@@ -3,3 +3,10 @@
   Memcheck:Cond
   fun:compare_data
 }
+{
+  Handle pthread/valgrind issue (possible leak)
+  Memcheck:Leak
+  fun:calloc
+  fun:calloc
+  fun:allocate_dtv
+}
```

Comment:
@lamm Why is this suppression needed?
Could you mention this in the commit message, please?

Reply:
Actually, the issue was that now all the threads were being joined.
Commit 39ad027 now fix this.